framework/src/base/ComputeNodalAuxVarsThread.C

```diff
@@ -51,13 +51,18 @@ ComputeNodalAuxVarsThread::onNode(ConstNodeRange::const_iterator & node_it)
 
   _fe_problem.reinitNode(node, _tid);
 
+  // Get a map of all active block restricted AuxKernel objects
+  const std::map<SubdomainID, std::vector<MooseSharedPointer<AuxKernel> > > & block_kernels = _storage.getActiveBlockObjects(_tid);
+
+  // Loop over all SubdomainIDs for the curnent node, if an AuxKernel is active on this block then compute it.
   const std::set<SubdomainID> & block_ids = _sys.mesh().getNodeBlockIds(*node);
   for (std::set<SubdomainID>::const_iterator block_it = block_ids.begin(); block_it != block_ids.end(); ++block_it)
   {
-    for (std::vector<AuxKernel*>::const_iterator aux_it = _auxs[_tid].activeBlockNodalKernels(*block_it).begin();
-         aux_it != _auxs[_tid].activeBlockNodalKernels(*block_it).end();
-         ++aux_it)
-      (*aux_it)->compute();
+    std::map<SubdomainID, std::vector<MooseSharedPointer<AuxKernel> > >::const_iterator iter = block_kernels.find(*block_it);
```

Comment:
We really should consider using unordered maps inside of these compute threaded areas.